test.js

```diff
@@ -1230,3 +1231,70 @@ describe('concurrency', function() {
     expect(ParallelApi.jobs).to.equal(17);
   });
 });
+
+describe('getBabelVersion()', function() {
+  it ('returns the correct version', function() {
+    var babelCorePath = path.join(__dirname, 'node_modules', 'babel-core', 'index.js');
+    var babelCorePackageJson = path.join(path.dirname(babelCorePath), 'package.json');
+    var packageJson = fs.readFileSync(babelCorePackageJson);
+    var expectedVersion = JSON.parse(packageJson).version;
+    expect(ParallelApi.getBabelVersion()).to.equal(expectedVersion);
+  });
+});
+
+describe('workerpool', function() {
+  var parallelApiPath = require.resolve('./lib/parallel-api');
+
+  var stringToTransform = "const x = 0;";
+
+  var options = {
+    inputSourceMap: false,
+    sourceMap: false,
+    plugins: [
+      'transform-strict-mode',
+      'transform-es2015-block-scoping'
+    ]
+  };
+
+  before(function() {
+    // shut down any workerpool that is running at this point
+    var babelCoreVersion = ParallelApi.getBabelVersion();
+    var something = 'v1/broccoli-babel-transpiler/workerpool/babel-core-' + babelCoreVersion;
+    var runningPool = process[something];
+    if (runningPool) {
+      return runningPool.terminate()
+        .then(function() {
+          delete process[something];
+        });
+    }
+  });
+
+  it('should limit to one pool per babel version', function() {
+    this.timeout(5*1000);
+    delete require.cache[parallelApiPath];
+    process.env.JOBS = '2';
+    var ParallelApiOne = require('./lib/parallel-api');
+    delete require.cache[parallelApiPath];
+    var ParallelApiTwo = require('./lib/parallel-api');
+
+    return Promise.all([
+      ParallelApiOne.transformString(stringToTransform, options),
+      ParallelApiOne.transformString(stringToTransform, options),
+      ParallelApiTwo.transformString(stringToTransform, options),
+      ParallelApiTwo.transformString(stringToTransform, options),
+    ]).then(function() {
+      return new Promise(function(resolve, reject) {
+        // find all worker processes
+        ps.lookup({
+          command: 'node',
+          arguments: path.join('broccoli-babel-transpiler', 'lib', 'worker.js'),
+        }, function(err, resultList ) {
+          expect(err).to.eql(null);
```

Comment:
lets `denodeify` ps.lookup, that way it is nicely promisified. Which will both simplify and improve the safety of our test.

```
let lookup = RSVP.denodeify(ps.lookup);
```

Reply:
ah good point, done